tests/src/Test/Tasty/Clash.hs

```diff
@@ -399,6 +425,63 @@ runTest1 modName opts@TestOptions{..} path target =
       (_, _, False) -> buildTests
       _ -> buildTests <> simTests
 
+  {-
+  buildAndSimTests name (buildTests, simTests)
+    -- Nothing
+    | isJust expectClashFail || not hdlLoad
+    = []
+
+    -- Build only
+    | not hdlSim
+    = [(name, testGroup name $ sequenceTests (show target : path) buildTests)]
+
+    -- Build and simulate
+    | otherwise
+    = [(name, testGroup name $ sequenceTests (show target : path) (buildTests <> simTests))]
+  -}
+
+  -- HACK: We want to run verilator and simulator tests independently if they
+  -- are both going to be run, otherwise failures from whichever comes first
+  -- will mean the second is skipped. In lieu of a better way to sequence tests
+  -- we can sequence tests for multiple simulators, then drop the first test
+  -- tree for all but the first simulator (as it will be copies of clash gen).
+  --
+  -- TODO: Since tasty doesn't provide one, we should really provide a better
+  -- set of combinators for describing test dependencies. That way we can have
+  -- some more principled way of having a test structure like
```

Comment:
It's definitely Tasty's weakest point. Basically we already work our way around it by passing down names so we can construct the absolute name of the test at test location. It would be some much more intuitive if you could just do something like: `TestGroupSeq [ ... ]`. But alas..